### PR TITLE
Fix ```x.py test``` failure when ```lldb -P``` fails

### PR DESCRIPTION
**Pull request by ghost**
I'm on windows and wanted to use ```-C linker=lld```, which meant I needed to install LLVM. (I think?) As soon as LLVM was added to my PATH, ```x.py test``` started failing on me with:
````
thread 'main' panicked at '"lldb" "-P" failed Output {
````
The error occurs just from LLVM being on the path, regardless of whether I actually use ```-C linker=lld```.

It turns out that ```src\bootstrap\test.rs``` attempts to call ```lldb -P``` in order to find some python directory and then pass it along via ```--lldb-python-dir```. If the command fails entirely due to e.g. ```lldb``` not being on the path, then this is accepted. However, if the command does run and returns an error exit code, this is not accepted.

I'm not sure why my ```lldb -P``` is failing (output below), but it seems to me that if the command missing entirely is tolerated, it's fine to also tolerate it exiting with an error.

For reference, my ```lldb -P``` error output:
````
PS C:\Users\skippy\rust> lldb -P
Python path configuration:
  PYTHONHOME = 'C:/Users/hwennborg/AppData/Local/Programs/Python/Python310'
  PYTHONPATH = (not set)
  program name = 'python'
  isolated = 0
  environment = 1
  user site = 1
  import site = 1
  sys._base_executable = 'C:\\Program Files\\LLVM\\bin\\lldb.exe'
  sys.base_prefix = 'C:/Users/hwennborg/AppData/Local/Programs/Python/Python310'
  sys.base_exec_prefix = 'C:/Users/hwennborg/AppData/Local/Programs/Python/Python310'
  sys.platlibdir = 'lib'
  sys.executable = 'C:\\Program Files\\LLVM\\bin\\lldb.exe'
  sys.prefix = 'C:/Users/hwennborg/AppData/Local/Programs/Python/Python310'
  sys.exec_prefix = 'C:/Users/hwennborg/AppData/Local/Programs/Python/Python310'
  sys.path = [
    'C:\\Users\\skippy\\AppData\\Local\\Programs\\Python\\Python310\\python310.zip',
    'C:/Users/hwennborg/AppData/Local/Programs/Python/Python310\\DLLs',
    'C:/Users/hwennborg/AppData/Local/Programs/Python/Python310\\lib',
    'C:\\Program Files\\LLVM\\bin',
  ]
Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
Python runtime state: core initialized
ModuleNotFoundError: No module named 'encodings'

Current thread 0x000007c8 (most recent call first):
  <no Python frame>
````

### CODE REVIEW

**Review by jyn514 — APPROVED** [review]
Your lldb error is very cursed, but the bootstrap changes look fine. It looks like lldb on your machine was generated with PyInstaller or something??

**Comment by skippy10110** [comment]
```winget install -e LLVM.LLVM``` install actually. The ````C:/Users/hwennborg/```` stood out to me as particularly odd, that's not me! 🤣

**Comment by skippy10110** [comment]
Looks like I'm hitting a new bug: llvm/llvm-project#54589

File: src/bootstrap/test.rs [hunk 1]
```diff
@@ -1418,14 +1418,10 @@ note: if you're sure you want to do this, please open an issue as to why. In the
             })
         };
         let lldb_exe = "lldb";
-        let lldb_version = Command::new(lldb_exe)
-            .arg("--version")
-            .output()
-            .map(|output| String::from_utf8_lossy(&output.stdout).to_string())
-            .ok();
+        let lldb_version = run(Command::new(lldb_exe).arg("--version"));
```
Comment (skippy10110) [inline src/bootstrap/test.rs:1421]: This ```--version``` change isn't strictly necessary, but I figured I'd make ```lldb --version``` consistent with ```lldb -P```. It seems like ```run``` is appropriate to use in both places?

My lldb is actually failing here too, so really it shouldn't even be trying to do ```lldb -P```, which it no longer would with this change.

I don't have a working lldb to test this.

**Comment by Mark-Simulacrum** [comment]
> I'm not sure why my lldb -P is failing (output below), but it seems to me that if the command missing entirely is tolerated, it's fine to also tolerate it exiting with an error.

I would disagree with this assertion. A missing command is a reasonable indication of "I don't have this tool", but a command exiting with an error to me sounds like "I installed it, but it doesn't actually work". I think failing in the latter case is a much better thing in general than silently ignoring the tool -- it means that e.g. if you have a permission error (no executable bit), or lldb changed its argument syntax and -P stops working, that will lead to our CI and generally user builds failing, rather than just no longer running some subset of tests.

I am inclined to close this as not working. I think the right step in your situation is probably to delete lldb from the LLVM install until that bug is fixed, or attempt to actually fix lldb.

**Comment by skippy10110** [comment]
In that case, it might make sense to keep the part where I made ```--version``` work the same way, as that error currently is silently ignored. But if it silently errors there, you're prolly about to blow up on ```-P``` anyway.

That's just nit picky cleanup without any real need, and I think I'm switching back to WSL for other reasons anyway, so I'm fine just closing this out.

**Pull request closed without merging**